engine/src/main/scala/org/scalarules/dsl/core/grammar/date/DslDate.scala

```diff
@@ -0,0 +1,16 @@
+package org.scalarules.dsl.core.grammar.date
+
+import org.joda.time.LocalDate
+
+/**
+  * Provides the DSL with a Date type. It supports the Ordered trait, which means it can be used
+  * in DslConditions with the comparative operators.
+  *
+  * @param internal internal representation of the date.
+  */
+case class DslDate private[date](internal: LocalDate) extends Ordered[DslDate] {
```

Comment:
I would actually consider calling the class 'LocalDate' and perhaps just place it in the org.scalarules.dsl.core package or one deeper (add .date or .time).
The reason for that is that it reads more intuitive and after all, both java 8 and JodaTime both call the class LocalDate as well, so there can be no name confusion.

For naming, in Scala you can give an Alias, so we can just import JodaTime date as JodaLocalDate